mipengine/controller/node_landscape_aggregator.py

```diff
@@ -144,39 +162,43 @@ def _update(self):
             try:
                 nodes_addresses = get_nodes_addresses()
                 nodes_info = _get_nodes_info(nodes_addresses)
-                with self._update_lock:
-                    self._node_registry.nodes = {
-                        node_info.id: node_info for node_info in nodes_info
-                    }
-                    local_nodes = [
-                        node for node in nodes_info if node.role == NodeRole.LOCALNODE
-                    ]
-
-                    (
-                        dataset_locations,
-                        aggregated_datasets,
-                    ) = _gather_all_dataset_info(local_nodes)
-                    data_model_cdes_per_node = _get_cdes_across_nodes(local_nodes)
-                    compatible_data_models = _get_compatible_data_models(
-                        data_model_cdes_per_node
-                    )
-                    _update_data_models_with_aggregated_datasets(
-                        data_models=compatible_data_models,
-                        aggregated_datasets=aggregated_datasets,
-                    )
-
-                    datasets_locations = (
-                        _get_dataset_locations_of_compatible_data_models(
-                            compatible_data_models, dataset_locations
-                        )
-                    )
-
-                    self._data_model_registry.data_models = compatible_data_models
-                    self._data_model_registry.datasets_location = datasets_locations
-
-                    logger.info(
-                        f"Online nodes:{[node for node in self._node_registry.nodes]}"
-                    )
+                local_nodes = {
```

Comment:
the lock is needed because it is a different thread that executes the `_update` method and a different thread that calls `get_nodes`, `get_global_node`, `get_all_local_nodes`, `get_node_info`, `get_cdes`, etc So if one of these methods is called while the `NodeRegistry` or the `DataModelRegistry` properties are altered inside `_update`(by a different thread) you do not really know what you re getting as a result, it can range from inaccurate information to corrupted data. Another alternative, would be instead of making the `NodeLandscapeAggregator` thread safe, put a lock in `NodeRegistry` and another lock inside `DataModelRegistry` and make them threadsafe.